filesystems/gcs/src/main/scala/cromwell/filesystems/gcs/GcsFileSystemProvider.scala

```diff
@@ -201,15 +204,37 @@ class GcsFileSystemProvider private (storageClient: Try[Storage], executionConte
   override def isHidden(path: Path): Boolean = throw new NotImplementedError()
 
   private def list(gcsDir: NioGcsPath) = {
-    val listRequest = client.objects().list(gcsDir.bucket)
+    val maxResults = ConfigFactory.load().getIntOr("google.list-max-results", 1000).toLong
```

Comment:
I agree with "meh, might as well", as a safety valve in case we suddenly need to change it in production... but I wouldn't bother documenting it.